### DISCUSSION TITLE
Eloquent cast to collection

### DISCUSSION THREAD

**Discussion by innocenzi** [spatie/laravel-data#238] [Ideas]
Hey,

Right now it's possible to cast a collection of data objects with the following syntax:

```php
protected $casts = [
    'contract_cancellation_terms' => DataCollection::class.':'.CancellationTermData::class
]
```

Would you be open to a PR implementing something along the lines of `CancellationTermData::castTo(DataCollection::class)`? 

Asking before PRing in order to get ideas on the syntax. I'd like a static method of some sort because the colon concatenation is not a nice syntax.

**Comment by rubenvanassche** [comment]
Hi @innocenzi,

If I'm not mistaken it is not allowed to use a static method in a constant expression like you're proposing here. My IDE provides me with a lot of red lines if I try something like this 😄

**Reply by innocenzi** [reply]
Aaah that is a very good point lol. Nevermind the suggestion then 😅

**Reply by rubenvanassche** [reply]
Yeah have made that mistake quite a lot myself 😅, congrats on the release of Hybridly btw

**Reply by innocenzi** [reply]
Thank you 😊
